### PR TITLE
Fix numpy bugs

### PR DESCRIPTION
**Pull request by reminisce**
## Description ##
This PR fixed several things.

1. Legacy `NDArray` `__setitem__` now works in the following case. @astonzhang 
```python
>>> import mxnet as mx
>>> a = mx.nd.ones((2, 3))
>>> b = mx.nd.zeros((1, 1, 1, 1, 1, 3))
>>> a[0] = b
>>> a

[[0. 0. 0.]
 [1. 1. 1.]]
<NDArray 2x3 @cpu(0)>
```

2. `np.true_divide` supports integral dtypes and outputs float32. @haojin2 

3. Loading legacy symbol json files in `np_shape` semantics. @stu1130 @frankfliu 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: src/operator/numpy/np_true_divide-inl.h [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        Kernel<mxnet_op::op_with_req<OP, Req>, xpu>::Launch(s, outputs[0].Size(),
+                                                            outputs[0].dptr<float>(),
+                                                            inputs[0].dptr<DType>(),
+                                                            inputs[1].dptr<DType>());
+      });
+    }
+  });
+}
+
+
```
Comment (haojin2) [inline src/operator/numpy/np_true_divide-inl.h]: get rid of the extra blank lines.

File: tests/python/unittest/test_symbol.py [hunk 2]
```diff
@@ -414,6 +416,47 @@ def test_gen_atomic_symbol_multiple_outputs():
...
+    batch_size = 10
+    num_hdidden = 128
+    num_features = 784
+
+    def get_net():
+        data = mx.sym.var('data')
+        weight = mx.sym.var('weight', shape=(num_hdidden, 0))
+        return mx.sym.FullyConnected(data, weight, num_hidden=num_hdidden)
+
+    for flag1 in [False, True]:
```
Comment (haojin2) [inline tests/python/unittest/test_symbol.py]: Maybe we need better names for `flag1` and `flag2`?

Comment (reminisce) [reply tests/python/unittest/test_symbol.py]: I will rename them in another PR to save the CI cycle for this one.

Comment (reminisce) [reply tests/python/unittest/test_symbol.py]: I will rename them in another PR to save the CI cycle for this one.

Comment (haojin2) [reply tests/python/unittest/test_symbol.py]: Okay, I'll resolve this one for now.

**Pull request merged**